server.js

```diff
@@ -4960,6 +4960,27 @@ cache(function(data, match, sendBadge, request) {
   });
 }));
 
+// Facebook integration.
+camp.route(/^\/facebook\/share\/([^\/]+)\/(.+)\.(svg|png|gif|jpg|json)$/,
+cache(function(data, match, sendBadge, request) {
+  var scheme = match[1]; // eg, https
+  var path = match[2];   // eg, shields.io
+  var format = match[3];
+  var page = encodeURIComponent(scheme + '://' + path);
+  var badgeData = getBadgeData('Share', data);
+  if (badgeData.template === 'social') {
+    badgeData.logo = badgeData.logo || logos.facebook;
+    badgeData.links = [
+      'https://www.facebook.com/sharer/sharer.php?u=' + page,
+      'https://www.facebook.com/search?q=' + page,
+     ];
+  }
+  badgeData.text[1] = '';
```

Comment:
Hmm. I mostly expect all badges to have a right-hand side. The only reason the Twitter URL badge doesn't is that it had one when it was implemented, but in an unexpected turn of event, Twitter removed support, breaking *all* Twitter buttons throughout the Web. Most of them looked really weird; I added a fallback on the Shields ones to avoid having a weird empty right-hand-side. I kept the endpoint to avoid breaking existing Shields badges in the wild.

However, a badge that doesn't actually fetch information is better served by the `/badge/` endpoint. For instance:

https://img.shields.io/badge/Share--white.svg?style=social&link=https%3A%2F%2Fwww.facebook.com%2Fsharer%2Fsharer.php%3Fu%3Dhttps%253A%252F%252Fshields.io&logo=data%3Aimage%2Fsvg%2Bxml%3Bbase64%2CPHN2ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHZpZXdCb3g9IjAgMCAzMzUgMjc2Ij48cGF0aCBmaWxsPSIjNTVhY2VlIiBkPSJNMzAyIDcwQTE5NSAxOTUgMCAwIDEgMyAyNDVhMTQyIDE0MiAwIDAgMCA5Ny0zMCA3MCA3MCAwIDAgMS01OC00NyA3MCA3MCAwIDAgMCAzMS0yIDcwIDcwIDAgMCAxLTU3LTY2IDcwIDcwIDAgMCAwIDI4IDUgNzAgNzAgMCAwIDEtMTgtOTAgMTk1IDE5NSAwIDAgMCAxNDEgNzIgNjcgNjcgMCAwIDEgMTE2LTYyIDExNyAxMTcgMCAwIDAgNDMtMTcgNjUgNjUgMCAwIDEtMzEgMzggMTE3IDExNyAwIDAgMCAzOS0xMSA2NSA2NSAwIDAgMS0zMiAzNSIvPjwvc3ZnPgo

(which can be automatically generated for convenience by suggest.js (which is called when inputting a URL in shields.io's search bar) and which can be shortened for convenience, eg. https://goo.gl/3PsfYi)

Woudl you prefer to contribute it to suggest.js, or would you prefer to add the right-hand-side by performing a call to Facebook's graphql?